lib/register.js

```diff
@@ -292,10 +292,15 @@
       else {
         module.dependencies.push(null);
       }
-
-      // run the setter for this dependency
-      if (module.setters[i])
-        module.setters[i](depExports);
+      
+      // run setters for all entries with the matching dependency name
+      var originalIndices = entry.originalIndices[i];
+      for (var j = 0, len = originalIndices.length; j < len; ++j) {
+        var index = originalIndices[j];
+        if (module.setters[index]) {
+          module.setters[index](depExports);
+        }
+      }
```

Comment:
Oh right I see, we're assuming the instantiate has to return deduped dependencies. Perhaps we can consider relaxing this constraint? Does it still work correctly that way?

Note that in the updated WhatWG spec, the instantiate result no longer needs to include dependencies, so we don't need to worry about what the spec does here actually.